core/src/impl/Kokkos_Error.hpp

```diff
@@ -194,6 +194,8 @@ KOKKOS_IMPL_ABORT_NORETURN KOKKOS_INLINE_FUNCTION void abort(
   Kokkos::Impl::hip_abort(message);
 #elif !defined(KOKKOS_ENABLE_OPENMPTARGET) && !defined(__HCC_ACCELERATOR__)
   Kokkos::Impl::host_abort(message);
+#else
+  (void)message;
```

Comment:
Actually we should not silent warnings in the else-clause because that could hide some hypothetical bug we would introduce.  What you really want is
```suggestion
#elif defined(KOKKOS_ENABLE_OPENMPTARGET)
  (void)message; // FIXME_OPENMPTARGET
```